delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphViewReferenceExchange.scala

```diff
@@ -0,0 +1,55 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.blazegraph
+
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.model._
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.ReferenceExchange
+import ch.epfl.bluebrain.nexus.delta.sdk.ReferenceExchange.ReferenceExchangeValue
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Event, ResourceRef}
+import monix.bio.{IO, UIO}
+
+/**
+  * BlazegraphView specific [[ReferenceExchange]] implementation.
+  *
+  * @param views the blazegraph module
+  */
+class BlazegraphViewReferenceExchange(views: BlazegraphViews) extends ReferenceExchange {
+
+  override type E = BlazegraphViewEvent
+  override type A = BlazegraphView
+
+  override def apply(
+      project: ProjectRef,
+      reference: ResourceRef
+  ): UIO[Option[ReferenceExchangeValue[BlazegraphView]]] =
+    reference match {
+      case ResourceRef.Latest(iri)           => resourceToValue(views.fetch(iri, project))
+      case ResourceRef.Revision(_, iri, rev) => resourceToValue(views.fetchAt(iri, project, rev))
+      case ResourceRef.Tag(_, iri, tag)      => resourceToValue(views.fetchBy(iri, project, tag))
+    }
+
+  private val schemaIri = model.schema.original
+
+  override def apply(
+      project: ProjectRef,
+      schema: ResourceRef,
+      reference: ResourceRef
+  ): UIO[Option[ReferenceExchangeValue[BlazegraphView]]] =
+    schema.original match {
```

Comment:
`schema.iri`
(ignores the rev or tag query param)

Reply:
it shouldn't ignore the tag or rev query params... because a schema reference can be revisioned or tagged